logger_loc_test.go

```diff
@@ -0,0 +1,59 @@
+package hclog
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// This file contains tests that are sensitive to their location in the file,
+// because they contain line numbers. They're basically "quarantined" from the
+// other tests because they break all the time when new tests are added.
```

Comment:
Would it be less brittle to assert the line matches a regex with `\d+` instead of a specific line number? that doesn't seem any less powerful to me assuming the log messages themselves are unique on ever line we output them which seems easy to achieve?

Reply:
Could, yes, but we actually want to validate the number itself, so we'd still be location dependent. Moving them to a different file at least isolates a common issue of them breaking when someone adds a test.